README.md

```diff
@@ -1,2 +1,401 @@
+
+1. Configure `west` to place build output in 
+   project-board-specific folders
+
+      ```bat
+      west config build.dir-fmt ${ZEPHYR_BASE}\..\build\{source_dir}\{board}
+      ```
+
+
+## Build and Run the zxa_example_project
+
+Build the `zxa_example_project` with
+`west build`, changing ``<your-board-name>``
+appropriately for your board:
+
+### Ubuntu
+
+1. Build projects directly from the command line
+
+      ```bash
+      cd ~/zxa
+      west build -p auto -b native_posix \
+        example-application/zephyr/projects/zxa_example_project
+      ```
+
+1. Run projects from the command line
+
+      ```bash
+      ./build/example-application/zephyr/projects/zxa_example_project/native_posix/zephyr/zephyr.exe
+      ```
+
+1. Build and run projects using Zephyr's `twister script`
+
+      ```bash
+      zephyr/scripts/twister \
+        -T example-application/zephyr/projects/zxa_example_project
+      ```
+
+1. Build and run the all projects (including projects and tests)
+   with code coverage using Zephyr's `twister script`
+
+      ```bash
+      zephyr/scripts/twister --coverage \
+        -T example-application/zephyr
+      ```
+
+### macOS
+
+1. Build projects directly from the command line
+
+      ```bash
+      cd ~/zxa
+      west build -p auto -b native_posix \
+        example-application/zephyr/projects/zxa_example_project
+      ```
+
+1. Run projects from the command line
+
+      ```bash
+      ./build/example-application/zephyr/projects/zxa_example_project/native_posix/zephyr/zephyr.exe
+      ```
+
+1. Build and run projects using Zephyr's `twister script`
+
+      ```bash
+      zephyr/scripts/twister \
+        -T example-application/zephyr/projects/zxa_example_project
+      ```
+
+1. Build and run the all projects (including projects and tests)
+   with code coverage using Zephyr's `twister script`
+
+      ```bash
+      zephyr/scripts/twister --coverage \
+        -T example-application/zephyr
+      ```
+
+### Windows
+
+1. Build projects directly from the command line
+
+      ```bat
+      cd %HOMEPATH%\zxa
+      west build -p auto -b native_posix \
+        example-application\zephyr\projects\zxa_example_project
+      ```
+
+1. Run projects from the command line
+
+      ```bat
+      .\build\example-application\zephyr\projects\zxa_example_project\native_posix\zephyr\zephyr.exe
+      ```
+
+1. Build and run projects using Zephyr's `twister script`
+
+      ```bat
+      zephyr\scripts\twister \
+        -T example-application\zephyr\projects\zxa_example_project
+      ```
+
+1. Build and run the all projects (including projects and tests)
+   with code coverage using Zephyr's `twister script`
+
+      ```bat
+      zephyr\scripts\twister --coverage \
+        -T example-application\zephyr
+      ```
+
+
+# Module Contribution Guidelines
+
+## Design Principles
+
+The ZXA module is designed around the following principles:
+
+- Keep all Zephyr-specific content within the (required) `zephyr` folder.
```

Comment:
We never intended for west projects that are also a manifest project to be architectured like this. This might be true for some modules, but for this one I do not think this is the right approach.

Reply:
It may not have been intended, but I have found it *extremely appropriate*, scalable, and working well for extending existing open source repositories, and makes for a very straight-forward separation of the Zephyr extensions from the other build system integrations and glue code already present.

Reply:
I think this is not the right approach because the whole purpose of this repository is to show how an out-of-tree **Zephyr** application with a driver and a board can be organized. This is not an example of how to take an existing repository and turn it into a zephyr module. Hence, the abstraction of having a single `zephyr/module.yml` that is the link between the repository as a module and the Zephyr build system but the rest of the content in the root of the repository is as simple as it gets, which is the goal here: simplicity. Users are typically overwhelmed with things to learn when they first come to Zephyr the last thing we want to do is overcomplicate this example for them.

Reply:
If we only have one example repository, then the example must be "all"-encompassing.  It sounds like you are targeting the person brand new to Zephyr, and I am targeting everyone above that. As far as v2.6.0 is concerned, the rich example would more thoroughly exercise the build system.